### PR TITLE
[SMALLFIX] Better upload configurations for S3

### PR DESCRIPTION
**Pull request by calvinjia**
Prevents the client from prematurely timing out and uses multipart upload more aggressively to saturate the upload to S3.

### CODE REVIEW

File: docs/_data/table/common-configuration.csv [hunk 1]
```diff
@@ -18,6 +18,9 @@ alluxio.underfs.hdfs.prefixes,"hdfs://,glusterfs:///"
 alluxio.underfs.s3.proxy.host,No default
 alluxio.underfs.s3.proxy.https.only,true
 alluxio.underfs.s3.proxy.port,No default
+alluxio.underfs.s3.max.threads,30
```
Comment (ifcharming) [inline docs/_data/table/common-configuration.csv]: Can you please add some comments to explain why we pick those magic numbers?

**Comment by ifcharming** [comment]
Overall LGTM, left one question.

File: docs/_data/table/common-configuration.csv (hunk 1 above)
Comment (calvinjia) [inline docs/_data/table/common-configuration.csv]: Updated, thanks. I also made it more consistent with jets3t's defaults.

**Comment by ifcharming** [comment]
LGTM

File: core/common/src/main/java/alluxio/Constants.java [hunk 2]
```diff
@@ -158,6 +158,10 @@
       "alluxio.underfs.s3.disable.dns.buckets";
   public static final String UNDERFS_S3_SERVER_SIDE_ENCRYPTION =
       "alluxio.underfs.s3.server.side.encryption";
+  public static final String UNDERFS_S3_MAX_THREADS = "alluxio.underfs.s3.max.threads";
```
Comment (apc999) [inline core/common/src/main/java/alluxio/Constants.java]: our convention is `alluxio.underfs.s3.threads.max`?

**Comment by apc999** [comment]
a high level question: it seems we keep adding supporting alluxio properties for jet3et in recent releases. Alternatively, we could simply forward all jet3et properties ?

**Comment by ifcharming** [comment]
+1 for @apc999 suggestion to support all jets3t properties. But in order to keep backwards-compatibility, do we want to limit the scope to be "Advanced" (or Alluxio-not-supported) jets3t properties only? otherwise setting Alluxio s3 properties might conflict with setting jets3t properties directly.

File: core/common/src/main/java/alluxio/Constants.java (hunk 2 above)
Comment (calvinjia) [inline core/common/src/main/java/alluxio/Constants.java]: Updated, thanks.

**Comment by ifcharming** [comment]
LGTM

File: core/common/src/main/java/alluxio/Constants.java [hunk 3]
```diff
@@ -158,6 +158,10 @@
       "alluxio.underfs.s3.disable.dns.buckets";
   public static final String UNDERFS_S3_SERVER_SIDE_ENCRYPTION =
       "alluxio.underfs.s3.server.side.encryption";
+  public static final String UNDERFS_S3_MAX_THREADS = "alluxio.underfs.s3.threads.max";
```
Comment (apc999) [inline core/common/src/main/java/alluxio/Constants.java]: could you change your constant name to be `UNDERFS_S3_THREADS_MAX` to be consistent ?

**Comment by apc999** [comment]
one minor comment let. thanks

File: core/common/src/main/java/alluxio/Constants.java (hunk 3 above)
Comment (calvinjia) [inline core/common/src/main/java/alluxio/Constants.java]: Updated, thanks.

**Comment by apc999** [comment]
LGTM

**Pull request merged**